lib/server.js

```diff
@@ -294,6 +294,50 @@ Server.prototype._onMetrics = function _onMetrics(req, callback) {
   }
 };
 
+Server.prototype._onTraceData = function _onTraceData(req, callback) {
+  debug('trace data: %j', req.trace);
+  var Process = this._app.models.ServiceProcess;
+  var AgentTrace = this._app.models.AgentTrace;
+
+  var q = {where: {serviceInstanceId: 1, pid: +req.processId, stopTime: null}};
+  Process.findOne(q, function(err, proc) {
+    assert.ifError(err);
+    if (!proc) {
+      debug('Ignoring trace for unknown process: %d', req.processId);
+      return callback();
+    }
+
+    async.each(req.trace.list,
+      saveTrace.bind(this, proc),
```

Comment:
Shouldn't `this` be null? It's not defined to anything at this point, and not used by saveTrace().